### PR TITLE
RDM-12549: Migrate internal search APIs to Access Profiles

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-12549

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java [hunk 1]
```diff
@@ -251,12 +251,14 @@
     @Mapping(source = "workBasketCaseFieldEntity.caseField.reference", target = "caseFieldId")
     @Mapping(source = "workBasketCaseFieldEntity.caseFieldElementPath", target = "caseFieldElementPath")
     @Mapping(source = "workBasketCaseFieldEntity.accessProfile.reference", target = "role")
+    @Mapping(source = "workBasketCaseFieldEntity.caseType.reference", target = "caseTypeId")
```
Comment (rafalkalita) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java:254]: Please update the EntityToResponseDTOMapperTest

**Review by rafalkalita — CHANGES_REQUESTED** [review]

File: domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java [hunk 2]
```diff
@@ -251,12 +251,14 @@
...
     @Mapping(expression = "java(workBasketCaseFieldEntity.getCaseField().isMetadataField())", target = "metadata")
     WorkBasketResultField map(WorkBasketCaseFieldEntity workBasketCaseFieldEntity);
 
     @Mapping(source = "searchCasesResultFieldEntity.caseField.reference", target = "caseFieldId")
     @Mapping(source = "searchCasesResultFieldEntity.caseFieldElementPath", target = "caseFieldElementPath")
     @Mapping(source = "searchCasesResultFieldEntity.accessProfile.reference", target = "role")
+    @Mapping(source = "searchCasesResultFieldEntity.caseType.reference", target = "caseTypeId")
```
Comment (danlysiak) [inline domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java:261]: Do we need it for `SearchResultsField` too?

Comment (kiran-yenigala-hmcts) [reply domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java:261]: ya.. just added that tooo

**Review by danlysiak — APPROVED** [review]
👍

**Review by rafalkalita — APPROVED** [review]

**Pull request merged**